pandas/tests/test_internals.py

```diff
@@ -566,6 +566,13 @@ def test_astype(self):
             else:
                 self.assertEqual(tmgr.get('d').dtype.type, t)
 
+    def test_illegal_arg_for_errors_in_astype(self):
+        """ ValueError exception raised when illegal value used for errors """
```

Comment:
I've looked at the existing test cases in `pandas/tests/series` & 
`pandas/tests/frame` and can't find a natural place to put this test.

I was thinking of adding a new file called say `test_astype.py` in
`pandas/tests/frame`, as there is an existing `test_apply.py`